src/validation.cpp

```diff
@@ -3809,7 +3809,12 @@ bool CChainState::DisconnectTip(CValidationState &state,
             mnview.GetHistoryWriters().DiscardDB();
             return error("DisconnectTip(): DisconnectBlock %s failed", pindexDelete->GetBlockHash().ToString());
         }
-        XResultThrowOnErr(evm_try_disconnect_latest_block(result));
+
+        if (auto res =
+                pcustomcsview->GetVMDomainBlockEdge(VMDomainEdge::DVMToEVM, pindexDelete->GetBlockHash().ToString())) {
```

Comment:
Would it be better to use the tryfrom XVM static method on the block coinbase tx to try to get the XVM struct information instead to determine if the DVM block has an EVM block?

Reply:
Sounds better, let me update

Reply:
LGTM 👍